src/EditorFeatures/Core.Wpf/Adornments/GraphicsResult.cs

```diff
@@ -12,22 +10,18 @@ namespace Microsoft.CodeAnalysis.Editor.Implementation.Adornments
     internal class GraphicsResult : IDisposable
     {
         public UIElement VisualElement { get; }
-        private Action _dispose;
+        private Action? _dispose;
 
-        public GraphicsResult(UIElement visualElement, Action dispose)
+        public GraphicsResult(UIElement visualElement, Action? dispose)
         {
             VisualElement = visualElement;
             _dispose = dispose;
         }
 
         public void Dispose()
         {
-            if (_dispose != null)
-            {
-                _dispose();
-
-                _dispose = null;
-            }
+            _dispose?.Invoke();
+            _dispose = null;
```

Comment:
If desired, there's a a thread-safe but slightly slower version:
```suggestion
            Interlocked.Exchange(ref _dispose, null)?.Invoke();
```

Reply:
Oh, you need `using System.Threading;` also